build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -204,31 +203,23 @@ class _SingleBuild {
     if (_outputMap != null && result.status == BuildStatus.success) {
       if (!await createMergedOutputDirectories(_outputMap, _assetGraph,
           _packageGraph, _reader, _environment, optionalOutputTracker)) {
-        result = _convertToFailure(
-            result, 'Failed to create merged output directories.',
-            failureType: FailureType.cantCreate);
+        result = _convertToFailure(result, failureType: FailureType.cantCreate);
       }
     }
     if (result.status == BuildStatus.success) {
       _logger.info('Succeeded after ${humanReadable(watch.elapsed)} with '
           '${result.outputs.length} outputs ($actionsCompletedCount actions)\n');
     } else {
-      if (result.exception is FatalBuildException) {
-        // TODO(???) Really bad idea. Should not set exit codes in libraries!
-        exitCode = 1;
```

Comment:
Lets add this to the changelog as well

Reply:
Discussed offline, will followup with a PR that removes the one other place we set `exitCode` and update the changelog.